src/Microsoft.CodeQuality.Analyzers/Core/ApiDesignGuidelines/DoNotRaiseExceptionsInUnexpectedLocations.cs

```diff
@@ -167,21 +167,25 @@ private static List<MethodCategory> GetMethodCategories(Compilation compilation)
             var methodCategories = new List<MethodCategory> {
                 new MethodCategory(IsPropertyGetter, true,
                     PropertyGetterRule,
-                    WellKnownTypes.InvalidOperationException(compilation), WellKnownTypes.NotSupportedException(compilation)),
+                    compilation.GetTypeByMetadataName(typeof(System.InvalidOperationException).FullName),
```

Comment:
Some types are used multiple time so I could create a local instance instead of doing the call to end up with the same instance. I am not sure whether this is a useless micro-optimization.